ctapipe/image/cleaning.py

```diff
@@ -56,8 +56,8 @@ def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
     else:
         # Require at least min_number_picture_neighbors. Otherwise, the pixel
         #  is not selected
-        number_of_neighbors_above_picture = np.sum(pixels_above_picture &
-                                                   geom.neighbor_matrix, axis=1)
+        number_of_neighbors_above_picture = geom.neighbor_matrix_sparse.dot(
+            pixels_above_picture.view(np.byte))
```

Comment:
regarding your previous comment, you could just add some note in the comment text here as to why the view is used (to coerce it into an int), so people won't be confused later